### PR TITLE
Check for Extra Input Field

### PR DESCRIPTION
**Pull request by sinhaashish**
Error out if any extra metadata field is passed as input

Code added to check extra meta fileds

## Description

The meta fields of policy is validated with the meta fields of input. 
Errors out If any extra meta field is present in input.

## Motivation and Context

[Fixes ](https://github.com/minio/minio/issues/6559)

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#806e3e8c-f354-4998-bf5b-d24646aea2b9) |
| mint-compression-dist-xl.sh | :x: [more...](#7e3a4834-63e5-4bca-be1e-3eaafe010732) |
<a name="806e3e8c-f354-4998-bf5b-d24646aea2b9"></a>
### 7437-dbe9025/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:31972
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp df08c65a82c3:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-c3cp9g1aiknu09",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 376,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.51:31972/aws-sdk-go-test-c3cp9g1aiknu09: dial tcp 72.28.97.51:31972: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="7e3a4834-63e5-4bca-be1e-3eaafe010732"></a>
### 7437-dbe9025/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:30069
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 40b08a131980:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-h3egsumy4dxedh",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 256,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.59:30069/aws-sdk-go-test-h3egsumy4dxedh: dial tcp 72.28.97.59:30069: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Comment by sinhaashish** [comment]
@bala @Praveenrajmani  PTAL

**Review by balamurugana — DISMISSED** [review]

File: cmd/postpolicyform.go [hunk 1]
```diff
@@ -222,6 +222,22 @@ func checkPostPolicy(formValues http.Header, postPolicyForm PostPolicyForm) erro
...
+		if strings.HasPrefix(cond, "$x-amz-meta-") {
+			formCanonicalName := http.CanonicalHeaderKey(strings.TrimPrefix(cond, "$"))
+			metaMap[formCanonicalName] = v.Value
+		}
+	}
+	// Check if any extra metadata field is passed as input
+	for key := range formValues {
+		if strings.HasPrefix(key, "X-Amz-Meta-") {
+			if _, ok := metaMap[key]; !ok {
+				return fmt.Errorf("Invalid according to Policy: Invalid according to Policy: Extra input fields: %s", key)
```
Comment (Praveenrajmani) [inline cmd/postpolicyform.go]: The Error message has to be corrected here

**Review by Praveenrajmani — CHANGES_REQUESTED** [review]

File: cmd/postpolicyform.go (hunk 1 above)
Comment (sinhaashish) [reply cmd/postpolicyform.go]: Done

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#2acced02-0981-4e8c-a23e-f92663f65a76) |
| mint-large-bucket.sh | :x: [more...](#fd33fd04-fdaf-47c7-9ca4-591a9ef5a11e) |
<a name="2acced02-0981-4e8c-a23e-f92663f65a76"></a>
### 7437-1a898f8/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:31347
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c4b51957b52f:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 47 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 27 seconds
(8/14) Running minio-dotnet tests ... done in 26 seconds
(9/14) Running minio-go tests ... done in 42 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 47 seconds
(11/14) Running minio-js tests ... done in 38 seconds
(12/14) Running minio-py tests ... FAILED in 3 minutes and 36 seconds
{
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "args": {
    "bucket_name": "minio-py-test-dbdb4ad6-a1b0-4026-affd-ae03d4a96472",
    "object_name": "d814b3e3-a41d-4772-b210-3c133a4c11a8"
  },
  "duration": 93,
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "status": "FAIL",
  "message": "AccessDenied: message: Access Denied",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1220, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1885, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1228, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n"
}

Executed 11 out of 14 tests successfully.
```
<a name="fd33fd04-fdaf-47c7-9ca4-591a9ef5a11e"></a>
### 7437-1a898f8/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:30349
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 3283d4f62b13:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-twlq302ol93c3r",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 3050,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:30349/aws-sdk-go-test-twlq302ol93c3r: dial tcp 72.28.97.57:30349: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Review by Praveenrajmani — APPROVED** [review]
Tested and LGTM

**Review by balamurugana — APPROVED** [review]

**Pull request merged**